lms/views/decorators/h_api.py

```diff
@@ -164,6 +164,33 @@ def wrapper(request, jwt):
     return wrapper
 
 
+def add_user_to_group(wrapped):
+    def wrapper(request, jwt):
+        get_param = functools.partial(_get_param, request)
+
+        tool_consumer_instance_guid = get_param("tool_consumer_instance_guid")
+        context_id = get_param("context_id")
+
+        group = models.CourseGroup.get(request.db, tool_consumer_instance_guid, context_id)
+        username = util.generate_username(request.params)
+        authority = request.registry.settings["h_authority"]
+        userid = f"acct:{username}@{authority}"
+        client_id = request.registry.settings["h_client_id"]
+        client_secret = request.registry.settings["h_client_secret"]
+
+        response = requests.post(
+            request.registry.settings["h_api_url"] + f"/groups/{group.pubid}/members/{userid}",
+            auth=(client_id, client_secret),
+            headers={
+                "X-Forwarded-User": userid,
+            },
+            timeout=1,
+        )
```

Comment:
No error handling here, and lots of duplication due to lack of centralised parameter valdation or an h API service etc. Still this is more or less how you would do it in a real version of this: on every LTI launch just send a member-join request for the user, regardless of whether they're already a member or not.